src/server/transaction.cc

```diff
@@ -931,12 +930,18 @@ void Transaction::ExecuteAsync() {
   DCHECK_GT(unique_shard_cnt_, 0u);
   DCHECK_GT(use_count_.load(memory_order_relaxed), 0u);
   DCHECK(!IsAtomicMulti() || multi_->lock_mode.has_value());
+  DCHECK_LE(shard_data_.size(), 128);
 
-  // Set armed flags on all active shards
-  IterateActiveShards([](auto& sd, auto i) { sd.local_mask |= ARMED; });
+  // Set armed flags on all active shards. Copy indices for dispatching poll tasks,
+  // because local_mask can be written concurrently after starting a new phase.
+  std::bitset<128> poll_flags(0);
```

Comment:
[we have today ](https://cloud.google.com/compute/docs/general-purpose-machines#n2d_machines)
"224 cpus". Lets at least aim for 1024.